src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,766 @@
+    {
+      ux_nm1[a] = ux_n[a];
+      uy_nm1[a] = uy_n[a];
+      uz_nm1[a] = uz_n[a];
+      ux_n[a] = ux_np1[a];
+      uy_n[a] = uy_np1[a];
+      uz_n[a] = uz_np1[a];
+
+      stiffnessVectorx[a] = 0.0;
+      stiffnessVectory[a] = 0.0;
+      stiffnessVectorz[a] = 0.0;
+      rhsx[a] = 0.0;
+      rhsy[a] = 0.0;
+      rhsz[a] = 0.0;
+    } );
+
+    for( real64 timeSeismo;
+         (timeSeismo = m_dtSeismoTrace*m_indexSeismoTrace) <= (time_n + epsilonLoc) && m_indexSeismoTrace < m_nsamplesSeismoTrace;
+         m_indexSeismoTrace++ )
+    {
+      /// do nothing
+    }
+
+  } );
+  return dt;
+
+}
+
+void ElasticWaveEquationSEM::cleanup( real64 const time_n, integer const, integer const, real64 const, DomainPartition & domain )
+{
+  // compute the remaining seismic traces, if needed
+  forMeshTargets( domain.getMeshBodies(), [&] ( string const &,
+                                                MeshLevel & mesh,
+                                                arrayView1d< string const > const & )
+  {
+    NodeManager & nodeManager = mesh.getNodeManager();
+    arrayView1d< real64 const > const ux_n = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementx_n >();
+    arrayView1d< real64 const > const ux_np1 = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementx_np1 >();
+    arrayView1d< real64 const > const uy_n = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementy_n >();
+    arrayView1d< real64 const > const uy_np1 = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementy_np1 >();
+    arrayView1d< real64 const > const uz_n = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementz_n >();
+    arrayView1d< real64 const > const uz_np1 = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementz_np1 >();
+    arrayView2d< real64 > const uxReceivers   = m_displacementxNp1AtReceivers.toView();
+    arrayView2d< real64 > const uyReceivers   = m_displacementyNp1AtReceivers.toView();
+    arrayView2d< real64 > const uzReceivers   = m_displacementzNp1AtReceivers.toView();
+
+    computeAllSeismoTraces( time_n, 0, ux_np1, ux_n, uxReceivers );
+    computeAllSeismoTraces( time_n, 0, uy_np1, uy_n, uyReceivers );
+    computeAllSeismoTraces( time_n, 0, uz_np1, uz_n, uzReceivers );
+  } );
+}
+
+void ElasticWaveEquationSEM::computeAllSeismoTraces( real64 const time_n,
+                                                     real64 const dt,
+                                                     arrayView1d< real64 const > const var_np1,
+                                                     arrayView1d< real64 const > const var_n,
+                                                     arrayView2d< real64 > varAtReceivers )
+{
+  localIndex indexSeismoTrace = m_indexSeismoTrace;
+
+  for( real64 timeSeismo;
+       (timeSeismo = m_dtSeismoTrace*m_indexSeismoTrace) <= (time_n + epsilonLoc) && m_indexSeismoTrace < m_nsamplesSeismoTrace;
```

Comment:
```suggestion
       (timeSeismo = m_dtSeismoTrace*indexSeismoTrace) <= (time_n + epsilonLoc) && indexSeismoTrace < m_nsamplesSeismoTrace;
```
Just replacing the `indexSeismoTrace`